src/problem3.rb

```diff
@@ -0,0 +1,9 @@
+class Problem3
+  def contains_duplicates(array)
+    unless array.nil?
+      array.uniq.length != array.length
```

Comment:
If you browse the [docs](http://ruby-doc.org/core-2.4.1/Enumerable.html#method-i-max) and you will click on toggle source you will see that `uniq` implementation uses a hash, the requirement of the problem is to use no additional data structures.